sessionctx/variable/session.go

```diff
@@ -1492,6 +1497,66 @@ func (s *SessionVars) AllocNewPlanID() int {
 	return s.PlanID
 }
 
+// Clear earliest CacheResult.
+func (s *SessionVars) clearEarliestCacheResult() {
+	var key uint32
+	minTime := time.Now()
+	for k, v := range s.cacheRes {
+		if v.LastUpdateTime.Before(minTime) {
+			minTime = v.LastUpdateTime
+			key = k
+		}
+	}
+	delete(s.cacheRes, key)
+}
+
+// SaveCache save result cache.
+func (s *SessionVars) SaveCache(cr *CacheResult) {
```

Comment:
When the client concurrency is high, the CacheResult is duplicated between different sessions and waste more memory.
Anyway, this is just a quick hack~

Reply:
IMO this should be fine...there will no high concurrency.